### PR TITLE
:sparkles: kustomize restricts about the scope of var definition

### PR DESCRIPTION
**Pull request by mengqiy**
We move var definition one level up to work around this restriction.
Kubebuilder now scaffolds kustomize v3.0.1+ and controller-gen v0.2.0-beta.3

### CODE REVIEW

File: pkg/scaffold/v2/certmanager/certificate.go [hunk 1]
```diff
@@ -52,10 +52,10 @@ metadata:
   name: serving-cert  # this name should match the one appeared in kustomizeconfig.yaml
   namespace: system
 spec:
-  # $(SERVICENAME) and $(NAMESPACE) will be substituted by kustomize
-  commonName: $(SERVICENAME).$(NAMESPACE).svc
+  # $(SERVICENAME) and $(SERVICENAMESPACE) will be substituted by kustomize
```
Comment (droot) [inline pkg/scaffold/v2/certmanager/certificate.go]: Can we use names with underscores to improve the readability ? SERVICENAME --> SERVICE_NAME

**Comment by mengqiy** [comment]
PTAL
Will squash commits before merging.

**Comment by mengqiy** [comment]
Squashed commits and rebased.
Reapplying label per https://github.com/kubernetes-sigs/kubebuilder/pull/852#event-2476859846

**Comment by mengqiy** [comment]
hmm... strange. Prow merged it already :confused:

**Pull request merged**